automation/devops_automation_infra/utils/s3.py

```diff
@@ -63,4 +74,16 @@ def download_file_to_filesystem(boto3_client, remote_path, local_dir=".", bucket
 
 def download_files_to_filesystem(boto3_client, remote_files, local_dir=".", bucketname="anyvision-testing"):
     for file in remote_files:
-        download_file_to_filesystem(boto3_client, file, local_dir, bucketname)
\ No newline at end of file
+        download_file_to_filesystem(boto3_client, file, local_dir, bucketname)
+
+
+def get_bucket_key_from_path(path):
+    full_path = path.replace('s3://', '/')
+    start = 0
+    comp = full_path.find('/', start)
+    if comp == 0:
+        start = 1
+        comp = full_path.find('/', start)
+    bucket = full_path[start: comp]
+    key = full_path[comp + 1:]
+    return bucket, key
```

Comment:
I think we could try and simplify the function, seems a bit complicated to understand and maintain.